abstractions/dotnet/src/serialization/ParsableFactory.cs

```diff
@@ -0,0 +1,15 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Kiota.Abstractions.Serialization;
+
+/// <summary>
+/// Defines the factory for creating parsable objects.
+/// </summary>
+/// <param name="node">The <see cref="IParseNode">node</see> to parse use to get the discriminator value from the payload.</param>
+/// <returns>The <see cref="IParsable">parsable</see> object.</returns>
+/// <typeparam name="T">The type of the parsable object.</typeparam>
+public delegate T ParsableFactory<T>(IParseNode node) where T : IParsable;
```

Comment:
Just to be clear, we are using `delegate` instead of `Func` because the generic parameter makes using `Func` trickier to use. Yes?

Reply:
yes and so it eases up maintenance on our side (ref counting, renaming, changing it here changes it everywhere, makes the type reusable for consumers etc...).
I didn't see any downside for the consumers (perf, or other aspects)